products/memgraph/upcoming/tutorials/understanding-music-with-modules.md

```diff
@@ -0,0 +1,450 @@
+and a list of genres he loved. The edges represent the mutual friendship between
+the users. You can find a more detailed explanation of the dataset on the
+[github](https://github.com/benedekrozemberczki/datasets#deezer-social-networks)
+alongside many more similar datasets kindly provided by the same authors.
+
+### Importing the Snapshot and defining a directory with user modules
+
+We have prepared a database snapshot for this example, so the user can easily
+import it when starting Memgraph using the `--data-directory` option.
+
+Also, we need to set up the directory from which the Memgraph will search for
+the query modules by changing the `--query-modules-directory` flag in the main
+configuration file(`/etc/memgraph/memgraph.conf`) or by supplying it as a
+command-line parameter using the directory containing our modules as the value.
+For more detailed explanation take a look at
+[How to Use and Implement Query Modules](../how_to_guides/use-and-implement-query-modules.md).
+
+```plaintext
+sudo -u memgraph \
+  /usr/lib/memgraph/memgraph --data-directory /usr/share/memgraph/examples/Europe \
+    --storage-snapshot-interval-sec=0 --storage-wal-enabled=false \
+    --storage-snapshot-on-exit=false --storage-properties-on-edges=true
+    --query-modules-directory=/path/to/your/modules/directory
+```
+
+When using Memgraph installed from DEB or RPM package, the currently running
+Memgraph server may need to be stopped before importing the example. The user
+can do so using the following command:
+
+```plaintext
+systemctl stop memgraph
+```
+
+When using Docker, the example can be imported with the following command
+(notice how we need to mount the query module directory first):
+
+```plaintext
+docker run -p 7687:7687 \
+  -v mg_lib:/var/lib/memgraph -v mg_log:/var/log/memgraph -v mg_etc:/etc/memgraph \
+  -v $(pwd)/path/to/your/modules/directory:/path/to/mount/of/modules/directory \
+  memgraph --data-directory /usr/share/memgraph/examples/Europe \
+  --storage-snapshot-interval-sec=0 --storage-wal-enabled=false \
+  --storage-snapshot-on-exit=false --storage-properties-on-edges=true
+  --query-modules-directory=/path/to/mount/of/modules/directory
+```
+
+The user should note that any modifications of the database state will persist
+only during this run of Memgraph.
+
+### Example Queries and procedures
+
+First, let's create a Python script in our modules directory which will contain
+our procedures. We'll name the script `deezer_example.py`.
+
+After each change to the script, we need to tell Memgraph to reload all the
+modules. We can do that by calling the following command:
+
+```opencypher
+CALL mg.load_all();
+```
+
+Let's count how many times a specific genre occurs in our network!
+
+We will define a procedure called `genre_count`:
+
+```python
+import mgp
+
+@mgp.read_proc
+def genre_count(context: mgp.ProcCtx,
+                genre: str
+               ) -> mgp.Record(genre=str, count=int):
```

Comment:
This violates `pep8 E124` rule, which style was used to format the code?

Reply:
Yeah, I noticed that too... Probably messed up somewhere when using the formatter.